src/add-return-info-csv-parser.js

```diff
@@ -0,0 +1,125 @@
+import { SphereClient } from 'sphere-node-sdk'
+import _ from 'underscore'
+import highland from 'highland'
+import JSONStream from 'JSONStream'
+import csv from 'csv-parser'
+import CONS from './constants'
+
+export default class AddReturnInfoCsvParser {
+  constructor (apiClientConfig, logger, config = {}) {
+    this.client = new SphereClient(
+      Object.assign(apiClientConfig, { user_agent: 'csv-parser-orders' })
+    )
+    this.logger = logger || {
+      error: process.stderr,
+      warn: process.stderr,
+      info: process.stdout,
+      verbose: process.stdout,
+    }
+
+    this.config = config
+    this.batchSize =
+      this.config.batchSize || CONS.standardOption.batchSize
+    this.delimiter =
+      this.config.delimiter || CONS.standardOption.delimiter
+    this.strictMode =
+      this.config.strictMode || CONS.standardOption.strictMode
+  }
+
+  parse (input, output) {
+    this.logger.verbose('Starting conversion')
+
+    let rowIndex = 1
+
+    highland(input)
+    .through(csv({
+      separator: this.delimiter,
+      strict: this.strictMode,
+    }))
+    .batch(this.batchSize)
+    .stopOnError(error => this.logger.error(error))
+    .doto((data) => {
+      this.logger.verbose(`Parsed row-${rowIndex}: ${JSON.stringify(data)}`)
+      rowIndex += 1
+    })
+    .flatMap(highland)
+    .flatMap(data => highland(this.processData(data, rowIndex)))
+    .stopOnError(error => this.logger.error(error))
+    .doto(data => this.logger.verbose(
+      `Converted row${rowIndex}: ${JSON.stringify(data)}`
+    ))
+    .reduce([], (memo, next) => {
```

Comment:
I have refactored this a bit to be more clear, the branch has two commits the first one makes the variable naming more clear. The second commit changes the loop structure.

https://github.com/commercetools/csv-parser-orders/pull/4/files